hub/api/dataset.py

```diff
@@ -0,0 +1,69 @@
+from hub.util.slice import merge_slices
+from typing import Union
+import numpy as np
+
+
+class Dataset:
```

Comment:
Would love to see some example code as well on how this will be used by the user.

Reply:
I'm working directly off of this:

https://activeloop.atlassian.net/wiki/spaces/OS/pages/128483386/API